Color/src/Graphics/Color/Space/OKLAB.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+-- |
+-- Module      : Graphics.Color.Space.OKLAB
+-- Copyright   : (c) Alexey Kuleshevich 2018-2020
+-- License     : BSD3
+-- Maintainer  : Alexey Kuleshevich <lehins@yandex.ru>
+-- Stability   : experimental
+-- Portability : non-portable
+--
+module Graphics.Color.Space.OKLAB
+  ( -- * Constructors for an CIE L*a*b* color space.
+    pattern OKLAB
+  , pattern ColorOKLAB
+  , pattern ColorOKLABA
+  , OKLAB
+  ) where
+
+import Foreign.Storable
+import Graphics.Color.Model.Internal
+import Graphics.Color.Space.Internal
+import Graphics.Color.Illuminant.ITU.Rec601 (D65)
+
+--------------
+--- OKLAB ---
+--------------
+
+-- | [OK L*a*b*](https://bottosson.github.io/posts/oklab/) color space
+data OKLAB
+
+-- | Color in OK L*a*b* color space
+newtype instance Color OKLAB e = OKLAB (V3 e)
+
+
+pattern ColorOKLAB :: e -> e -> e -> Color OKLAB e
+pattern ColorOKLAB l' a' b' = OKLAB (V3 l' a' b')
+{-# COMPLETE ColorOKLAB #-}
+
+-- | Constructor for @OKLAB@ with alpha channel.
+pattern ColorOKLABA :: e -> e -> e -> e -> Color (Alpha OKLAB) e
+pattern ColorOKLABA l' a' b' a = Alpha (OKLAB (V3 l' a' b')) a
+{-# COMPLETE ColorOKLABA #-}
+
+-- | `OKLAB` color space
+deriving instance Eq e => Eq (Color OKLAB e)
+
+-- | `OKLAB` color space
+deriving instance Ord e => Ord (Color OKLAB e)
+
+-- | `OKLAB` color space
+deriving instance Functor (Color OKLAB)
+
+-- | `OKLAB` color space
+deriving instance Applicative (Color OKLAB)
+
+-- | `OKLAB` color space
+deriving instance Foldable (Color OKLAB)
+
+-- | `OKLAB` color space
+deriving instance Traversable (Color OKLAB)
+
+-- | `OKLAB` color space
+deriving instance Storable e => Storable (Color OKLAB e)
+
+-- | `OKLAB` color space
+instance (Elevator e) => Show (Color OKLAB e) where
+  showsPrec _ = showsColorModel
+
+-- | `OKLAB` color space
+instance (Elevator e) => ColorModel OKLAB e where
+  type Components OKLAB e = (e, e, e)
+  toComponents (ColorOKLAB l' a' b') = (l', a', b')
+  {-# INLINE toComponents #-}
+  fromComponents (l', a', b') = ColorOKLAB l' a' b'
+  {-# INLINE fromComponents #-}
+
+instance (Elevator e, RealFloat e) => ColorSpace OKLAB D65 e where
+  type BaseModel OKLAB = OKLAB
+  type BaseSpace OKLAB = OKLAB
+  toBaseSpace = id
+  {-# INLINE toBaseSpace #-}
+  fromBaseSpace = id
+  {-# INLINE fromBaseSpace #-}
+  luminance c = Y $ lab2luminance c
+  {-# INLINE luminance #-}
+  toColorXYZ = lab2xyz
+  {-# INLINE toColorXYZ #-}
+  fromColorXYZ = xyz2lab
+  {-# INLINE fromColorXYZ #-}
+
+lab2luminance :: forall a e. (Elevator e, Elevator a, RealFloat e, RealFloat a)
+  => Color OKLAB e
+  -> a
+lab2luminance (ColorOKLAB ll a b) = toRealFloat y
+  where
+    !l' = 1.0000000000000000 * ll + 0.3963377773761749 * a + 0.2158037573099136 * b
+    !m' = 1.0000000000000000 * ll - 0.1055613458156586 * a - 0.0638541728258133 * b
+    !s' = 1.0000000000000000 * ll - 0.0894841775298119 * a - 1.2914855480194092 * b
```

Comment:
Multiplication by 1 is superfluous.
```suggestion
    !l' = ll + 0.3963377773761749 * a + 0.2158037573099136 * b
    !m' = ll - 0.1055613458156586 * a - 0.0638541728258133 * b
    !s' = ll - 0.0894841775298119 * a - 1.2914855480194092 * b
```
The only reason why multiplication by 1 is done in the css color spec, is because they define the matrix and then use matrix multiplication functionality.

Reply:
Done.